lib/MTT/Values/Functions/MPI/OMPI.pm

```diff
@@ -167,19 +167,18 @@ sub get_version {
 
     my $match = qr/ompi:version:full:(.*)$/;
 
-    if (not open INFO, "$bindir/ompi_info --parsable|") {
-        open INFO, "$bindir/ompi_info -V|";
+    my @out = `$bindir/ompi_info --parsable`;
+    if ($? >> 8) {
+        @out = `$bindir/ompi_info -V`;
```

Comment:
missing check if ompi_info succeeded

Reply:
Don't need it. Next loop will match nothing and return it. Very backward compatible :)

Reply:
but no next loop here? it is inside "if"
what do I miss

Reply:
`for` for `@out` is outside this `if`, closing `}` is on line 174.